addons/osio-addon/src/main/java/io/fabric8/launcher/osio/OsioEmptyLaunchMissionControl.java

```diff
@@ -0,0 +1,88 @@
+package io.fabric8.launcher.osio;
+
+import javax.enterprise.context.Dependent;
+import javax.inject.Inject;
+
+import io.fabric8.launcher.core.api.Boom;
+import io.fabric8.launcher.core.api.ImmutableBoom;
+import io.fabric8.launcher.core.api.MissionControl;
+import io.fabric8.launcher.core.api.Projectile;
+import io.fabric8.launcher.core.api.events.StatusMessageEventBroker;
+import io.fabric8.launcher.core.api.projectiles.CreateProjectile;
+import io.fabric8.launcher.core.api.projectiles.context.CreateProjectileContext;
+import io.fabric8.launcher.osio.client.OsioWitClient;
+import io.fabric8.launcher.osio.client.Space;
+import io.fabric8.launcher.osio.projectiles.ImmutableOsioLaunchProjectile;
+import io.fabric8.launcher.osio.projectiles.OsioLaunchProjectile;
+import io.fabric8.launcher.osio.projectiles.context.OsioProjectileContext;
+import io.fabric8.launcher.osio.steps.GitSteps;
+import io.fabric8.launcher.osio.steps.OpenShiftSteps;
+import io.fabric8.launcher.osio.steps.WitSteps;
+import io.fabric8.launcher.service.git.api.GitRepository;
+import io.fabric8.launcher.service.openshift.api.ImmutableOpenShiftProject;
+import io.fabric8.openshift.api.model.BuildConfig;
+
+
+@Dependent
+public class OsioEmptyLaunchMissionControl implements MissionControl<OsioProjectileContext, OsioLaunchProjectile> {
+
+    @Inject
+    private MissionControl<CreateProjectileContext, CreateProjectile> missionControl;
+
+    @Inject
+    private GitSteps gitSteps;
+
+    @Inject
+    private OpenShiftSteps openShiftSteps;
+
+    @Inject
+    private WitSteps witSteps;
+
+    @Inject
+    private OsioWitClient witClient;
+
+    @Inject
+    private StatusMessageEventBroker eventBroker;
+
+    @Override
+    public OsioLaunchProjectile prepare(OsioProjectileContext context) {
+        final Projectile projectile = missionControl.prepare(context);
+        final Space space = witClient.findSpaceById(context.getSpaceId())
+                .orElseThrow(() -> new IllegalStateException("Context space not found: " + context.getSpaceId()));
+        return ImmutableOsioLaunchProjectile.builder()
+                .from(projectile)
+                .gitOrganization(context.getGitOrganization())
+                .space(space)
+                .eventConsumer(eventBroker::send)
+                .pipelineId(context.getPipelineId())
+                .build();
+    }
+
+    @Override
+    public Boom launch(OsioLaunchProjectile projectile) {
+        // Make sure that cd-github is created in Openshift
+        openShiftSteps.ensureCDGithubSecretExists();
+
+        final GitRepository repository = gitSteps.createRepository(projectile);
```

Comment:
Nothing is being pushed to the repository, is that intended?

Reply:
yes, we need repository with no content.

Reply:
In this case why bother calling prepare and choosing a specific mission? This doesn't make much sense to me.